### ISSUE TITLE
Support HTTP2 out of the box

### ISSUE THREAD

**Issue by davewasmer** [denali-js/denali#95]

**Comment by acorncom** [comment]
Looks like main two options here are https://www.npmjs.com/package/spdy and https://www.npmjs.com/package/http2

The first seems to be more popular

**Comment by davewasmer** [comment]
Been doing some research and spiking out here.

I swapped out `http` for `spdy`, but I'm unable to get the server to respond - I get "Empty response from server" every time, or I get a response but via HTTP 1.1.

I did notice that most HTTP2 tutorials start with setting up SSL. It appears that the spec does _not_ require SSL, but no major browser supports plaintext HTTP2. I'm hitting my local server with curl, and no luck. @acorncom any chance you have any experience with HTTP2?

**Comment by acorncom** [comment]
@davewasmer I don't have much experience with HTTP2 yet, but I'd be happy to take a look.  Do you happen to have a branch I can checkout to take a look?

**Comment by davewasmer** [comment]
Nope, but the changes were super easy - just replace the use of `http` in `lib/runtime/application.js` with `spdy`.

**Comment by acorncom** [comment]
I'm seeing the same results (HTTP/1.1 working fine still but HTTP/2 sending back an empty response when SSL is disabled and the `plain` option for `spdy` is turned on).  I'm inclined to assume that `curl` is expecting SSL to be enabled ...

For future reference, these are good instructions on how to get curl to run with http2: https://simonecarletti.com/blog/2016/01/http2-curl-macosx/

My resulting curl does properly check out against Cloudflare with the following command:

`/usr/local/bin/curl -I --http2 https://www.cloudflare.com/`

So I think we'll have to get our SSL story worked out before HTTP2 becomes something we can enable ...

**Comment by acorncom** [comment]
Confirmed that we need SSL by running `nghttp -nv http://localhost:3000` `nghttp` was mentioned in this article (https://blog.cloudflare.com/tools-for-debugging-testing-and-using-http-2/) and worked fine against CloudFlare.  It also mentions negotiating the h2 protocol at the beginning of its frame listings.  Doesn't display the same message against `spdy` without a cert ...

But we *could* enable spdy as our default server and allow folks to send in options (including SSL certs) and whether they wanted SSL ...  That way we've got it "available" out of the box even if we don't have it easily setup in dev ...

**Comment by acorncom** [comment]
Depends on #96
